
<launch>
    <arg name="venv" value="/home/quentin/Dev/argos3-ros-triangulation/venv/bin/python3" />

    <arg name="input_dir" value="/home/quentin/Dev/iridia-dwm/src/ros_workspace/src/static_convergence/script/output/raw_data" />
    <arg name="input_file" value="raw_data_robot_4_try_3_delay_0_batch_5" />
    <arg name="simulation" value="True" />

    <arg name="output_dir" value="/home/quentin/Dev/iridia-dwm/src/ros_workspace/src/static_convergence/script/output/convergence" />
    <arg name="output_file" value="static_convergence_robot_4_try_3_delay_0_batch_5" />

    <arg name="experiment_duration" default="60" />

    <node pkg="ros_utils" type="experiment_manager.py" name="experiment_manager" output="screen" cwd="node" launch-prefix="$(arg venv)" args="$(arg experiment_duration)" ></node>

    <!-- TODO: put sensor reading in another ROS package -->
    <node pkg="static_convergence" type="sensor_measurements.py" name="sensor_measurements" output="screen" cwd="node" launch-prefix="$(arg venv)" args="$(arg simulation) $(arg input_dir) $(arg input_file)" ></node>

    <arg name="n" default="4" />
    <arg name="b" default="fbB" />
    <arg name="beacons" default="Z" />

    <!-- fbB computation and statistics -->
    <node pkg="static_convergence" type="main.py" name="main" output="screen" cwd="node" args="$(arg b) $(arg n) $(arg beacons)" launch-prefix="$(arg venv)"></node>
    <node pkg="static_convergence" type="statistics.py" name="statistics" output="screen" cwd="node" args="$(arg output_dir) $(arg output_file) $(arg n) $(arg b)" launch-prefix="$(arg venv)" ></node>

</launch>